pkg/pf/tests/internal/providerbuilder/build_provider.go

```diff
@@ -130,5 +141,18 @@ func NewProvider(params NewProviderArgs) *Provider {
 		}
 	}
 
+	for i := range prov.AllDataSources {
+		d := &prov.AllDataSources[i]
+		if d.DataSourceSchema.Attributes == nil {
+			d.DataSourceSchema.Attributes = map[string]dschema.Attribute{}
+		}
+
+		if d.ReadFunc == nil {
+			d.ReadFunc = func(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+				resp.State = tfsdk.State(req.Config)
+			}
+		}
```

Comment:
Since we make sure it's never `nil` here. We should probably drop one of these. I'm not sure I agree with the default also.
```suggestion
```

Reply:
Do you have a suggestion of a better default?